entry_types/scrolled/package/src/frontend/Entry.js

```diff
@@ -16,13 +19,22 @@ export default withInlineEditingDecorator('EntryDecorator', function Entry(props
 
   const entryStructure = useEntryStructure();
 
+  const previousSectionIndex = usePrevious(currentSectionIndex);
+
+  const sectionChangeEventEmitter = useSectionChangeEventEmitter(entryStructure);
+
   const setCurrentSectionIndex = useCallback(index => {
-    if (window.parent !== window) {
-      window.parent.postMessage({type: 'CHANGE_SECTION', payload: {index}}, window.location.origin);
+    if (index !== previousSectionIndex) {
+      sectionChangeMessagePoster(index);
+      sectionChangeEventEmitter(index);
+      setCurrentSectionIndexState(index);
     }
-
-    setCurrentSectionIndexState(index);
-  }, [setCurrentSectionIndexState]);
+  }, [
+    previousSectionIndex,
+    sectionChangeEventEmitter,
+    sectionChangeMessagePoster,
+    setCurrentSectionIndexState
+  ]);
```

Comment:
`sectionChangeMessagePoster` is an extra dependency since it is not from scope of the component's function. There should be a linter warnings about this in your editor.

This change makes the `useCallback` no-op since `sectionChangeEventEmitter` is a new function every time.

This still emits multiple events when the section changes. React does not apply state changes synchronously. So `setCurrentSectionIndex` can be called multiple times while `index` is still the old value.

You need to wrap the code in an effect to make it run after state changes have been applied:

```
function useSectionChangeEvents(entryStructure, currentSectionIndex) {
  const previousSectionIndex = usePrevious(currentSectionIndex);

  useEffect(() => {
    if (previousSectionIndex !== currentSectionIndex) {
      events.trigger('page:change', ...)
    }
  }, [entryStructure, previousSectionIndex, currentSectionIndex]);
}
```

For the `CHANGE` post message, we currently do not care about duplicate messages. So you can leave it unchanged for now if you want.

Reply:
I don't understand the first paragraph of your comment. There is no linter warning on `sectionChangeMessagePoster`. When I change that reference into an invalid one, e.g. `sectionChangeMessagePoster**s**`, there *is* a linter warning about an unnecessary dependency, so my linter-editor integration works in principle

Reply:
This is what I see:
```
$ yarn run eslint src/frontend/Entry.js                                                                                                  
yarn run v1.22.0
$ /home/tfischbach/code/gh/pageflow/node_modules/.bin/eslint src/frontend/Entry.js

/home/tfischbach/code/gh/pageflow/entry_types/scrolled/package/src/frontend/Entry.js
  32:6  warning  React Hook useCallback has an unnecessary dependency: 'sectionChangeMessagePoster'. Either exclud
e it or remove the dependency array. Outer scope values like 'sectionChangeMessagePoster' aren't valid dependencie
s because mutating them doesn't re-render the component  react-hooks/exhaustive-deps                             

```

Reply:
Interesting. My system eslint is version 7.3.1, as opposed to the 6.x versions that this branch currently uses. Perhaps my editor uses system eslint then. I'll look into this after responding to the remainder of your comment

Reply:
So, as soon as I removed the event-related code from the callback, I got the warning you mentioned about one of the two remaining mentioned dependencies. Still, I never observed exactly what you observed. It shall remain a mystery